cmake/config/CMakeLists.txt

```diff
@@ -219,4 +221,163 @@ FOREACH(_name ${_deal_ii_features_sorted})
   ENDFOREACH()
 ENDFOREACH()
 
+
+########################################################################
+#                                                                      #
+#                           pkgconfig files                            #
+#                                                                      #
+########################################################################
+
+#
+# Set up the pkgconfig configuration files consisting of
+#
+#   deal.II_base       # common part to all the rest
+#   deal.II_debug      # generated when buildtype is debug
+#   deal.II_release    # generated when buildtype is release
+#
+# We support two configurations out of which deal.II can be used - directly
+# from the build directory or after installation. So we have to prepare
+# two distinct setups.
+#
+# pkgconfig looks for *.pc files in an environmental variable called
+# PKG_CONFIG_PATH. So, to use the library in the build directory issue
+# export PKG_CONFIG_PATH=/path/to/BUILD_DIR/lib/pkgconfig:$PKG_CONFIG_PATH
+#
+# To use the library in the installed location
+# export PKG_CONFIG_PATH=/path/to/INSTALL_DIR/lib/pkgconfig:$PKG_CONFIG_PATH
+#
+
+#
+# Build up the link line from our list of libraries:
+#
+
+FOREACH(_build ${DEAL_II_BUILD_TYPES})
+  STRING(TOLOWER ${_build} _build_lowercase)
+
+  SET(_name "${DEAL_II_BASE_NAME}${DEAL_II_${_build}_SUFFIX}")
+
+  SET(CONFIG_RPATH_${_build}_PC "")
+  SET(CONFIG_LIBRARIES_${_build}_PC "-l${_name}")
+
+  FOREACH(_lib ${DEAL_II_LIBRARIES_${_build}} ${DEAL_II_LIBRARIES})
+
+    GET_FILENAME_COMPONENT(_name ${_lib} NAME)
+    GET_FILENAME_COMPONENT(_dir ${_lib} PATH)
+
+    IF("${_dir}" STREQUAL "")
+      # ${_lib} is a simple library name, just add it to the link line:
+      SET(_library_string "-l${_lib}")
+
+    ELSE()
+      # ${_lib} is a full library path:
+
+      #
+      # Only append a library directory if it is new ...
+      #
+
+      list (FIND CONFIG_RPATH_${_build}_PC "${_dir}" _index)
+      IF (${_index} EQUAL -1)
+        SET(_library_string "-L${_dir} ")
+        LIST(APPEND CONFIG_RPATH_${_build}_PC ${_dir})
+      ELSE()
+        SET(_library_string "")
+      ENDIF()
+
+      # Recover short name:
+      STRING(REGEX REPLACE
+        "^(${CMAKE_STATIC_LIBRARY_PREFIX}|${CMAKE_SHARED_LIBRARY_PREFIX})"
+        "" _name "${_name}"
+        )
+      STRING(REGEX REPLACE
+        "(${CMAKE_STATIC_LIBRARY_SUFFIX}|${CMAKE_SHARED_LIBRARY_SUFFIX})$"
+        "" _name "${_name}"
+        )
+      SET(_library_string "${_library_string}-l${_name}")
+    ENDIF()
+
+    SET(CONFIG_LIBRARIES_${_build}_PC
+      "${CONFIG_LIBRARIES_${_build}_PC} ${_library_string}"
+      )
+  ENDFOREACH()
+
+  TO_STRING_AND_ADD_PREFIX(CONFIG_RPATH_${_build}_PC
+    "-Wl,-rpath," ${CONFIG_RPATH_${_build}_PC}
```

Comment:
We could guard the rpath setup for the install directory with `CMAKE_INSTALL_RPATH_USE_LINK_PATH` (this would also require to refactor the one rpath line in the base pkgconf file).

Reply:
I guess its best to fully respect the stipulated configuration wrt. rpath. I think that I use different settings on different machines, so I could imagine this breaking for one of those cases.